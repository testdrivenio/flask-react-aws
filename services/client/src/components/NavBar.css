.link:hover {
  cursor: pointer;
  background-color: #292929;
  color: #f5f5f5;
}
