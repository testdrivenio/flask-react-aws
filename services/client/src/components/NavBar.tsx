import React from "react";
import {
  Box,
  Flex,
  Heading,
  Link,
  Spacer,
  IconButton,
  Stack,
  Collapse,
  useDisclosure,
} from "@chakra-ui/react";
import { HamburgerIcon, CloseIcon } from "@chakra-ui/icons";
import { Link as RouterLink } from "react-router-dom";

interface NavBarProps {
  title: string;
  logoutUser: () => void;
  isAuthenticated: () => boolean;
}

const NavBar: React.FC<NavBarProps> = ({
  title,
  logoutUser,
  isAuthenticated,
}) => {
  const { isOpen, onToggle } = useDisclosure();

  return (
    <Box bg="gray.800" px={4}>
      <Flex h={16} alignItems="center" justifyContent="space-between">
        <Flex alignItems="center">
          <Heading as="h1" size="lg" className="navbar-item nav-title" mr={4}>
            <Link
              as={RouterLink}
              to="/"
              _hover={{ textDecoration: "none" }}
              color="white"
            >
              {title}
            </Link>
          </Heading>
        </Flex>
        <Spacer />
        <Flex alignItems="center" display={{ base: "none", md: "flex" }}>
          <NavLinks logoutUser={logoutUser} isAuthenticated={isAuthenticated} />
        </Flex>
        <IconButton
          display={{ base: "flex", md: "none" }}
          onClick={onToggle}
          icon={isOpen ? <CloseIcon /> : <HamburgerIcon />}
          variant="outline"
          aria-label="Toggle Navigation"
          color="white"
        />
      </Flex>

      <Collapse in={isOpen} animateOpacity>
        <Stack mt={2} display={{ md: "none" }} bg="gray.800" p={4} spacing={4}>
          <NavLinks logoutUser={logoutUser} isAuthenticated={isAuthenticated} />
        </Stack>
      </Collapse>
    </Box>
  );
};

const NavLinks: React.FC<{
  logoutUser: () => void;
  isAuthenticated: () => boolean;
}> = ({ logoutUser, isAuthenticated }) => (
  <>
    <Link as={RouterLink} to="/about" mr={4} color="white">
      About
    </Link>
    {isAuthenticated() ? (
      <>
        <Link as={RouterLink} to="/status" mr={4} color="white">
          User Status
        </Link>
        <Link
          as="span"
          onClick={logoutUser}
          mr={4}
          color="white"
          cursor="pointer"
        >
          Log Out
        </Link>
      </>
    ) : (
      <>
        <Link as={RouterLink} to="/register" mr={4} color="white">
          Register
        </Link>
        <Link as={RouterLink} to="/login" mr={4} color="white">
          Log In
        </Link>
      </>
    )}
  </>
);

export default NavBar;
