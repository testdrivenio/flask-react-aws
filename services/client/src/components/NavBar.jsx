import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./NavBar.css";

const titleStyle = {
  fontWeight: "bold"
};

const NavBar = props => {
  let menu = (
    <div className="navbar-menu">
      <div className="navbar-start">
        <Link to="/about" className="navbar-item" data-testid="nav-about">
          About
        </Link>
      </div>
      <div className="navbar-end">
        <Link to="/register" className="navbar-item" data-testid="nav-register">
          Register
        </Link>
        <Link to="/login" className="navbar-item" data-testid="nav-login">
          Log In
        </Link>
      </div>
    </div>
  );
  if (props.isAuthenticated()) {
    menu = (
      <div className="navbar-menu">
        <div className="navbar-start">
          <Link to="/about" className="navbar-item" data-testid="nav-about">
            About
          </Link>
          <Link to="/status" className="navbar-item" data-testid="nav-status">
            User Status
          </Link>
        </div>
        <div className="navbar-end">
          <span
            // eslint-disable-next-line react/jsx-handler-names
            onClick={props.logoutUser}
            className="navbar-item link"
            data-testid="nav-logout"
          >
            Log Out
          </span>
        </div>
      </div>
    );
  }
  return (
    <nav
      className="navbar is-dark"
      role="navigation"
      aria-label="main navigation"
    >
      <section className="container">
        <div className="navbar-brand">
          <Link to="/" className="navbar-item nav-title" style={titleStyle}>
            {props.title}
          </Link>
          <span
            className="nav-toggle navbar-burger"
            onClick={() => {
              let toggle = document.querySelector(".nav-toggle");
              let menu = document.querySelector(".navbar-menu");
              toggle.classList.toggle("is-active");
              menu.classList.toggle("is-active");
            }}
          >
            <span />
            <span />
            <span />
          </span>
        </div>
        {menu}
      </section>
    </nav>
  );
};

NavBar.propTypes = {
  title: PropTypes.string.isRequired,
  logoutUser: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.func.isRequired
};

export default NavBar;
