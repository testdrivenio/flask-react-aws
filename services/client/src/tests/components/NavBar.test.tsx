import React from "react";
import { render, screen } from "../test-utils";
import { it, expect } from "vitest";
import "@testing-library/jest-dom/vitest";

import NavBar from "../../components/NavBar";

const mockProps = {
  title: "Hello, World!",
  logoutUser: () => {},
  isAuthenticated: () => false,
};

it("NavBar renders without crashing", () => {
  render(<NavBar {...mockProps} />);

  const titleElement = screen.getByText("Hello, World!");
  expect(titleElement).toBeInTheDocument();
  expect(titleElement.closest("h1")).toHaveClass("navbar-item");
  expect(titleElement.closest("h1")).toHaveClass("nav-title");
});

it("NavBar renders properly", () => {
  const { asFragment } = render(<NavBar {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
