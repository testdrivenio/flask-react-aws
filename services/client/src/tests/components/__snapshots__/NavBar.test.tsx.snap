// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar renders properly 1`] = `
<DocumentFragment>
  <div
    class="css-1ggpwrh"
  >
    <div
      class="css-19pkskf"
    >
      <div
        class="css-70qvj9"
      >
        <h1
          class="chakra-heading navbar-item nav-title css-1x64xs0"
        >
          <a
            class="chakra-link css-2s4j8o"
            href="/"
          >
            Hello, World!
          </a>
        </h1>
      </div>
      <div
        class="css-17xejub"
      />
      <div
        class="css-xv0w4j"
      >
        <a
          class="chakra-link css-fmj561"
          href="/about"
        >
          About
        </a>
        <a
          class="chakra-link css-fmj561"
          href="/register"
        >
          Register
        </a>
        <a
          class="chakra-link css-fmj561"
          href="/login"
        >
          Log In
        </a>
      </div>
      <button
        aria-label="Toggle Navigation"
        class="chakra-button css-3cjmcm"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: none; opacity: 0; height: 0px;"
    >
      <div
        class="chakra-stack css-12jrb58"
      >
        <a
          class="chakra-link css-fmj561"
          href="/about"
        >
          About
        </a>
        <a
          class="chakra-link css-fmj561"
          href="/register"
        >
          Register
        </a>
        <a
          class="chakra-link css-fmj561"
          href="/login"
        >
          Log In
        </a>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
